integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXUpgradeTests.java

```diff
@@ -63,14 +90,34 @@ public void testDistroXUpgrade(TestContext testContext) {
                 .withTemplate(String.format(commonClusterManagerProperties.getInternalDistroXBlueprintType(), currentRuntimeVersion))
                 .withPreferredSubnetsForInstanceNetworkIfMultiAzEnabledOrJustFirst()
                 .when(distroXTestClient.create(), key(distroXName))
-                .await(STACK_AVAILABLE)
+                .validate();
+        createImageValidationSourceCatalog(testContext, commonClusterManagerProperties.getUpgrade()
+                .getImageCatalogUrl3rdParty(), thirdPartyCatalogName);
+        uuid.set(getUuid(testContext, thirdPartyCatalogName, currentRuntimeVersion3rdParty));
```

Comment:
Oh, so the `getUuid()` invocation is not inside of a lambda? Maybe I was delusional seeing it in a lambda last time. :smile: Anyway, with the present approach, it is not even necessary to employ an `AtomicReference<String>`; `uuid` could be a regular `String`. (I suggested the `AtomicReference` as I thought the value would be calculated & set by a lambda.)